lib/queuePopulator/IngestionProducer.js

```diff
@@ -0,0 +1,210 @@
+const async = require('async');
+const bucketclient = require('bucketclient');
+const Logger = require('werelogs').Logger;
+const MetadataWrapper = require('arsenal').storage.metadata.MetadataWrapper;
+
+// const config = require('../../conf/Config');
+const constants = require('../../constants');
+const RaftLogEntry = require('../../extensions/utils/RaftLogEntry');
+
+class IngestionProducer {
+    /**
+     * Create an IngestionProducer class that helps create a snapshot of
+     * pre-existing RING backend
+     *
+     * @constructor
+     * @param {object} sourceConfig - source config with value for
+     *                                bucketdBootstrap
+     */
+    constructor(sourceConfig) {
+        this.log = new Logger('Backbeat:IngestionProducer');
+        this.ringParams = {
+            bucketdBootstrap: [sourceConfig.host],
+            bucketdLog: undefined,
+        };
+        this.ringReader = new MetadataWrapper('scality', this.ringParams,
+            bucketclient, this.log);
+        this.requestLogger = this.log.newRequestLogger();
+        this.createEntry = new RaftLogEntry();
+        this.resLog = [];
+        console.log('IngestionProducer has been created');
+    }
+
+    snapshot(raftId, done) {
+        async.waterfall([
+            next => this._getBuckets(raftId, next),
+            (bucketList, next) => this._getBucketMd(bucketList, next),
+            (bucketList, next) => this._getBucketObjects(bucketList, next),
+            (bucketList, next) =>
+                this._getBucketObjectsMetadata(bucketList, next),
+        ], err => done(err, this.resLog));
```

Comment:
Don't you need to re-initialize `this.resLog` before returning its previous value?

Not sure it should be done here as I'm not sure what is the lifetime of `resLog` in the class, but I think there should be some point where it's reinitialized, unless the class instance lifetime ends after `snapshot()` is called?